plugins/events-backend/src/service/http/HttpPostIngressEventPublisher.ts

```diff
@@ -108,32 +102,17 @@ export class HttpPostIngressEventPublisher {
     const logger = this.logger;
 
     router.post(path, async (request, response) => {
-      const requestBody = request.body;
-      if (!Buffer.isBuffer(requestBody)) {
-        throw new Error(
-          `Failed to retrieve raw body from incoming event for topic ${topic}; not a buffer: ${typeof requestBody}`,
-        );
-      }
-
-      const bodyBuffer: Buffer = requestBody;
-      const parsedContentType = contentType.parse(request);
-      if (
-        !parsedContentType.type ||
-        parsedContentType.type !== 'application/json'
-      ) {
-        throw new UnsupportedMediaTypeError(parsedContentType.type);
-      }
+      const requestContentType = contentType.parse(request);
+      const bodyParser = this.bodyParsers[requestContentType.type];
 
-      const encoding = parsedContentType.parameters.charset ?? 'utf-8';
-      if (!Buffer.isEncoding(encoding)) {
-        throw new UnsupportedCharsetError(encoding);
+      if (!bodyParser) {
+        throw new UnsupportedMediaTypeError(requestContentType.type);
       }
 
-      const bodyString = bodyBuffer.toString(encoding);
-      const bodyParsed =
-        parsedContentType.type === 'application/json'
-          ? JSON.parse(bodyString)
-          : bodyString;
+      const { bodyParsed, bodyBuffer, encoding } = await bodyParser(
+        request,
+        topic,
+      );
```

Comment:
We could pass the parsed content type here as well. This would remove the need to parse it inside of `bodyParser(..)` again.
From the interface perspective, it seems not necessary, though.
The decision could depend on how heavy or lightweight the parsing is.